File: С/caesar/Caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
  //right key?
  if (argc !== 2)
  {
    printf("Usage: ./caesar kay\n");
    return 1;
  }

  for (int i = 0, a = strlen(argv[1]); i < a; i++)
  {
    if (!argv[1][i] >= '0' && !argv[1][i] <= '9')
    {
      printf("Usage: ./caesar kay\n");
      return 1;
    }
  }

  int kay = atoi(argv[1]);
  int n;
  string s = get_string("Plaintext: ");
  printf("Ciphertext: ");

  for (int i = 0, int len = strlen(s); i < len; i++)
  {
    //small letter? - encrypt
    if (s[i] >= 'a' && s[i] <= 'z')
    {
      n = (int) s[i] + kay - 97;
      n = n %26;
      n = n + 97;
      printf("%c", n);
    }
    //capital letter? - encrypt
    else if (s[i] >= 'A' && s[i] <= 'Z')
    {
        n = (int) s[i] + kay - 65;
        n = n %26;
        n = n + 65;
        printf("%c", n);
    }
    //another character - not            
    else 
    {
      printf("%c", s[i]);
    }
  }
  printf("\n");
  return 0;
}

File: Python/HomeWork/caesar/caesar.py
# повністю робоча

from cs50 import get_string
from sys import argv
import sys

if len(argv) == 2:
    key = int(argv[1])

    plaintext = get_string("plaintext: ")

    # масив з цифровими значеннями літер тексту
    unicodetext = []
    for c in plaintext:
        char = ord(c)
        unicodetext.append(char)

    # масив з зашифрованими цифровими значеннями літер тексту
    cipherunicodetext = []
    for c in unicodetext:

        # формуля для
        if c < 91 and c > 64:
            n = (c - 64 + key) % 26
            n = 64 + n
            cipherunicodetext.append(n)
        elif c < 123 and c > 96:
            n = (c - 96 + key) % 26
            n = 96 + n
            cipherunicodetext.append(n)
        else:
            cipherunicodetext.append(c)


    # масив літер зашифрованого тексту
    ciphertext = []
    for c in cipherunicodetext:
        char = chr(c)
        ciphertext.append(char)
        
    # вивід зашифрованого тексту
    print("ciphertext: ", end = "")
    
    for c in ciphertext:
        print(c, end = "")
    print()
else:
    print("Usage: python caesar.py k")
    sys.exit(1)